core/src/main/java/org/apache/carbondata/core/scan/collector/impl/DictionaryBasedVectorResultCollector.java

```diff
@@ -51,8 +51,8 @@
 
   public DictionaryBasedVectorResultCollector(BlockExecutionInfo blockExecutionInfos) {
     super(blockExecutionInfos);
-    queryDimensions = tableBlockExecutionInfos.getQueryDimensions();
-    queryMeasures = tableBlockExecutionInfos.getQueryMeasures();
+    queryDimensions = tableBlockExecutionInfos.getActualQueryDimensions();
```

Comment:
I did not get, if any restructuring is required then it supposed to go to `RestructureBasedVectorResultCollector` class right. why it comes here.